libraries/aws_s3_buckets.rb

```diff
@@ -0,0 +1,108 @@
+require '_aws'
+
+class AwsS3Buckets < Inspec.resource(1)
+  name 'aws_s3_buckets'
+  desc 'Verifies settings for AWS S3 Buckets in bulk'
+  example "
+    describe aws_s3_buckets do
+      its('buckets.all') { should be_in ['logging_bucket', 'another_bucket'] }
+      it { should_not have_public_buckets }
+    end
+  "
+  include AwsResourceMixin
+  attr_reader :table, :buckets, :has_public_buckets
+  alias have_public_buckets? has_public_buckets
+  alias has_public_buckets? has_public_buckets
+
+  # Underlying FilterTable implementation.
+  filter = FilterTable.create
+  filter.add_accessor(:where)
+        .add_accessor(:entries)
+        .add(:exists?) { |x| !x.entries.empty? }
+        .add(:names, field: :name)
+  filter.connect(self, :table)
+
+  def to_s
+    'S3 Buckets'
+  end
+
+  private
+
+  def validate_params(raw_params)
+    validated_params = check_resource_param_names(
+      raw_params: raw_params,
+      allowed_params: [],
+      allowed_scalar_name: nil,
+      allowed_scalar_type: String,
+    )
+    validated_params
+  end
+
+  def fetch_from_aws
+    # Transform into filter format expected by AWS
+    filters = []
+    [
+      :table,
+      :buckets,
+      :has_public_buckets,
+    ].each do |criterion_name|
+      val = instance_variable_get("@#{criterion_name}".to_sym)
+      next if val.nil?
+      filters.push(
+        {
+          name: criterion_name.to_s.tr('_', '-'),
+          values: [val],
+        },
+      )
+    end
+    @table   = []
+    @buckets = Hashie::Mash.new({})
+    %w{
+      public all
+    }.each { |bucket| @buckets[bucket] ||= [] }
+    backend  = AwsS3Buckets::BackendFactory.create
+    # Note: should we ever implement server-side filtering
+    # (and this is a very good resource for that),
```

Comment:
Copypasta comment.  You can't do serverside filtering on `list_buckets`.  Please remove.